src/clj_puppetdb/core.clj

```diff
@@ -55,9 +53,9 @@
   contains additional metadata. Currently the only supported kind of metadata is:
   - :total (the total number of records available)"
   ([client path params]
-    (query-with-metadata client path nil params))
+   (query-with-metadata client path nil params))
   ([client path query-vec params]
-   (let [merged-params (merge params
+   (let [merged-params (merge {} params
```

Comment:
If both ```params``` and ```query-vec``` are ```nil``` then the result is ```nil``` to and that is rejected at some level by a precondition stipulating that the it must be a ```map```.